src/ophyd_async/core/detector.py

```diff
@@ -309,7 +311,7 @@ async def complete(self):
         assert self._arm_status, "Prepare not run"
         assert self._trigger_info
         async for index in self.writer.observe_indices_written(
-            self._frame_writing_timeout
+            self._frame_writing_timeout + self._trigger_info.livetime
```

Comment:
This does not cover the use case of trigger -> trigger -> trigger -> long wait -> trigger -> trigger -> trigger, as the long wait may still be longer than `frame_writing_timeout + livetime`. I think we need a new variable inside `trigger_info`, unless the frame writing timeout is set per call to prepare

Reply:
Re-reading #327 I can see the confusion. 
For _most_ use cases, we want the timeout to be `DEFAULT_TIMEOUT + self._trigger_info.livetime + self._trigger_info.deadtime` (i.e. the exposure time plus a default timeout). This makes sense for a time-resovled acquisition taking x frames at y fps. _However_, in this case we have an arbitrary extra timeout because we are waiting for a user to press a button and we don't know exactly how long that will take. @timsnow suggests 60s is a reasonable timeout. So the suggestion is that we pass in an arbitrary, optional, per-plan timeout via trigger info. If (and only if) we do this, the detector will use that timeout. If we don't pass it the detector defaults to `DEFAULT_TIMEOUT + self._trigger_info.livetime + self._trigger_info.deadtime` (this is probably the majority use case).

So the trigger info should have a new attribute `frame_timeout: float | None = None` (so that shouldn't break anything else in ophyd-async). This line should then be:

```suggestion
            self._trigger_info.frame_timeout or (DEFAULT_TIMEOUT + self._trigger_info.livetime + self._trigger_info.deadtime)
```

And we can remove `self._frame_writing_timeout` altogether because it is no longer used.

Tagging @abbiemery for visibility.

Reply:
I'll give that a go and see how it looks/.

Reply:
Could we not do the same thing with just the exisiting `self._frame_writing_timeout`? If we provide a value on instantiation then use it, otherwise, `(DEFAULT_TIMEOUT + self._trigger_info.livetime + self._trigger_info.deadtime)`

Reply:
I may have missed something, take a look. I do think it needs a specific tests for if the timeout is provided in the `TriggerInfo`. https://github.com/bluesky/ophyd-async/pull/348